Signal/src/AppDelegate.m

```diff
@@ -156,12 +156,15 @@ - (UIViewController *)loadingRootViewController
         [[UIStoryboard storyboardWithName:@"Launch Screen" bundle:nil] instantiateInitialViewController];
 
     BOOL shouldShowUpgradeLabel = NO;
-    NSString *lastAppVersion = AppVersion.instance.lastAppVersion;
     NSString *lastCompletedLaunchAppVersion = AppVersion.instance.lastCompletedLaunchAppVersion;
-    BOOL mayNeedUpgrade = ([TSAccountManager isRegistered] &&
-        [VersionMigrations isVersion:lastAppVersion atLeast:@"2.0.0" andLessThan:@"2.13.0"]);
+    NSString *kLastVersionWithDatabaseViewChange = @"2.13.0";
+    BOOL mayNeedUpgrade = ([TSAccountManager isRegistered]
+        && (!lastCompletedLaunchAppVersion ||
+               [VersionMigrations isVersion:lastCompletedLaunchAppVersion
+                                    atLeast:@"2.0.0"
+                                andLessThan:kLastVersionWithDatabaseViewChange]));
     BOOL hasCompletedUpgrade = (lastCompletedLaunchAppVersion &&
```

Comment:
nit: Can't we remove `hasCompletedUpgrade` all together since there is already a `andLessThan` check in `mayNeedUpgrade`?

If so it might make sense to rename `mayNeedUpgrade` -> `needsUpgrade`